dbt_af/conf/config.py

```diff
@@ -207,6 +208,78 @@ class K8sConfig:
     airflow_identity_binding_selector: str = attrs.field(default=None)
 
 
+@attrs.define(frozen=True)
+class RetryPolicy:
+    """
+    Config to configure airflow retries for one type of tasks
+    """
+
+    retries: int | None = None
+    retry_delay: datetime.timedelta | None = None
+    retry_exponential_backoff: bool | None = None
+    max_retry_delay: datetime.timedelta | float | None = None
+
+    def as_dict(self) -> dict[str, Any]:
+        return {
+            _attr.name: getattr(self, _attr.name)
+            for _attr in attrs.fields(RetryPolicy)
+            if getattr(self, _attr.name) is not None
+        }
+
+
+@attrs.define(frozen=True)
+class RetriesConfig:
+    """
+    Config to parametrize different retry policies for all dbt-af dag components.
+    Default policy must be configured or used with default values.
+    Default policy will be used if task-specific policy is not specified.
+    """
+
+    default_retry_policy: RetryPolicy = attrs.field(
+        default=RetryPolicy(
+            retries=2,
+            retry_delay=datetime.timedelta(minutes=1),
+            retry_exponential_backoff=False,
+        )
+    )
+
+    # dbt specific tasks
+    dbt_run_retry_policy: RetryPolicy = attrs.field(default=None)
+    dbt_seed_retry_policy: RetryPolicy = attrs.field(default=None)
+    dbt_snapshot_retry_policy: RetryPolicy = attrs.field(default=None)
+    dbt_test_retry_policy: RetryPolicy = attrs.field(default=None)
+    dbt_source_freshness_retry_policy: RetryPolicy = attrs.field(default=None)
+
+    # dbt-af specific tasks
+    sensor_retry_policy: RetryPolicy = attrs.field(default=None)
+    k8s_task_retry_policy: RetryPolicy = attrs.field(default=None)
+    macros_retry_policy: RetryPolicy = attrs.field(default=None)
+    supplemental_task_retry_policy: RetryPolicy = attrs.field(default=None)
+
+    def __attrs_post_init__(self):
+        """
+        Substitute all unspecified policies with default one.
+        If some values in task-specific policies are not specified, they will be used from default policy
+        """
+        for _attr in attrs.fields(RetriesConfig):
+            _attr: attrs.Attribute
+            if _attr.name == 'default_retry_policy':
+                continue
+
+            policy = getattr(self, _attr.name)
+            policy: RetryPolicy | None
+            # if a policy is completely unset, it's overwritten with default policy
+            if policy is None:
+                object.__setattr__(self, _attr.name, self.default_retry_policy)
+                continue
+
+            # if some parameters of the policy are not set, they are overwritten with values from default policy
+            for _policy_attr in attrs.fields(RetryPolicy):
+                _policy_attr: attrs.Attribute
+                if getattr(policy, _policy_attr.name) is None:
+                    object.__setattr__(policy, _policy_attr.name, getattr(self.default_retry_policy, _policy_attr.name))
```

Comment:
Don't like this. Can't we make it simpler?
Move `default_retry_policy` to global constant
And use fair default for every field?

Reply:
`default_retry_policy` is not a constant. You can still specify a default retry policy for your project, but you can also customize other policies at the same time.